### DISCUSSION TITLE
What function is CB_ENSURE()?

### DISCUSSION THREAD

**Discussion by Wangpc-972** [catboost/catboost#2948] [Q&A] [answered]
catboost version:0.25

When I browse the source code, there are many such functions:
```
CB_ENSURE()
CB_ENSURE_INTERNAL()
CB_ENSURE_SCALE_IDENTITY ()
```
What is their function? Where can I find these functions? Thank you!

**Comment by grag90** [comment, marked as answer]
These functions are our variations of assert. You could find their code in ` catboost/catboost/libs/helpers/exception.h ` and `catboost/catboost/libs/model/scale_and_bias.h`.

**Comment by Wangpc-972** [comment]
> These functions are our variations of assert. You could find their code in `catboost/catboost/libs/helpers/exception.h` and `catboost/catboost/libs/model/scale_and_bias.h`.

Thank you for your reply！！！
